File: 프로그래머스/1/68935. 3진법 뒤집기/3진법 뒤집기.py
def solution(n):
    answer = 0
    list_3nd = []
    
    while n:
        remain = n % 3
        list_3nd.append(remain)
        n //= 3
        
    for i in range(len(list_3nd)):
        answer += list_3nd[i] * 3**(len(list_3nd)-i-1)
        
    return answer

File: 프로그래머스/1/68935. 3진법 뒤집기/3진법 뒤집기.java
import java.util.*;

class Solution {
    public int solution(int n) {
        int answer = 0;
        String threeDemical = convertThreeDemical(n);
        for(int i = 0; i < threeDemical.length(); i++){
            answer += Character.getNumericValue(threeDemical.charAt(i)) * Math.pow(3, i);
        }
        
        return answer;
    }
    
    public String convertThreeDemical(int n){
        StringBuilder sb = new StringBuilder();
        while(n > 0){
            sb.append(n%3);
            n /= 3;
        }
        return sb.reverse().toString();
    }
}